File: C++/d) Heap Dynamic.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int size;
    cout << "Enter array size: ";
    cin >> size;

    // Heap dynamic array using vector - can grow/shrink
    vector<int> arr;

    cout << "Enter " << size << " elements:" << endl;
    for(int i = 0; i < size; i++) {
        int val;
        cin >> val;
        arr.push_back(val);
    }

    cout << "Heap dynamic array in C++:" << endl;
    for(int num : arr) {
        cout << num << " ";
    }
    cout << endl;

    // No manual cleanup needed for vector
    return 0;
}

File: Java/d) Heap Dynamic.java
import java.util.ArrayList;
import java.util.Scanner;

public class HeapDynamic {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter initial array size: ");
        int size = scanner.nextInt();
        
        // Heap dynamic array using ArrayList - can grow/shrink
        ArrayList<Integer> arr = new ArrayList<>();
        
        System.out.println("Enter " + size + " elements:");
        for(int i = 0; i < size; i++) {
            arr.add(scanner.nextInt());
        }
        
        System.out.println("Heap dynamic array in Java:");
        for(int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
        
        scanner.close();
    }
}